xchange-latoken/src/main/java/org/knowm/xchange/latoken/LatokenExchange.java

```diff
@@ -0,0 +1,148 @@
+import org.knowm.xchange.latoken.service.LatokenAccountService;
+import org.knowm.xchange.latoken.service.LatokenMarketDataService;
+import org.knowm.xchange.latoken.service.LatokenTradeService;
+import org.knowm.xchange.utils.AuthUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import si.mazi.rescu.RestProxyFactory;
+import si.mazi.rescu.SynchronizedValueFactory;
+
+public class LatokenExchange extends BaseExchange {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LatokenExchange.class);
+  private static final int PRECISION = 8;
+
+  public static final String sslUri = "https://api.latoken.com";
+
+  @Override
+  protected void initServices() {
+
+    this.marketDataService = new LatokenMarketDataService(this);
+    this.tradeService = new LatokenTradeService(this);
+    this.accountService = new LatokenAccountService(this);
+  }
+
+  /** Latoken uses HMAC signature and timing-validation to identify valid requests. */
+  @Override
+  public SynchronizedValueFactory<Long> getNonceFactory() {
+
+    throw new UnsupportedOperationException(
+        "Latoken uses HMAC signature and timing-validation rather than a nonce");
+  }
+
+  @Override
+  public ExchangeSpecification getDefaultExchangeSpecification() {
+
+    ExchangeSpecification spec = new ExchangeSpecification(this.getClass().getCanonicalName());
+    spec.setSslUri(sslUri);
+    spec.setHost("www.latoken.com");
+    spec.setPort(80);
+    spec.setExchangeName("Latoken");
+    spec.setExchangeDescription("LATOKEN Exchange.");
+    AuthUtils.setApiAndSecretKey(spec, "latoken");
+    return spec;
+  }
+
+  @Override
+  public void remoteInit() {
+
+    try {
+      // Load the static meta-data and override with the dynamic one
+      Map<Currency, CurrencyMetaData> currenciesMetaData = exchangeMetaData.getCurrencies();
+      Map<CurrencyPair, CurrencyPairMetaData> pairsMetaData = exchangeMetaData.getCurrencyPairs();
+
+      Latoken latokenPublic =
+          RestProxyFactory.createProxy(LatokenAuthenticated.class, LatokenExchange.sslUri);
+      List<LatokenPair> allPairs = latokenPublic.getAllPairs();
+      List<LatokenCurrency> allCurrencies = latokenPublic.getAllCurrencies();
+
+      // Allow adapters to use pairs data for future symbols adapting.
+      LatokenAdapters.setAllPairs(allPairs);
+
+      // Update Currency meta-data
+      for (LatokenCurrency latokenCurrency : allCurrencies) {
+        Currency currency = LatokenAdapters.adaptCurrency(latokenCurrency);
+        addCurrencyMetadata(currenciesMetaData, currency, PRECISION);
+      }
+
+      // Update CurrencyPair meta-data
+      for (LatokenPair latokenPair : allPairs) {
+        CurrencyPair pair = LatokenAdapters.adaptCurrencyPair(latokenPair);
+        CurrencyPairMetaData pairMetadata = LatokenAdapters.adaptPairMetaData(latokenPair);
+        addCurrencyPairMetadata(pairsMetaData, pair, pairMetadata);
+      }
+    } catch (LatokenException | IOException e) {
```

Comment:
this catch seams redundant, the error will get logged twice, here and in the application code
just leave the second catch